crates/ibc/src/core/ics03_connection/error.rs

```diff
@@ -7,168 +7,88 @@ use crate::signer::SignerError;
-                    e.target_height, e.oldest_height)
-            },
-
-        InvalidIdentifier
-            [ ValidationError ]
-            | _ | { "identifier error" },
-
-        EmptyProtoConnectionEnd
-            | _ | { "ConnectionEnd domain object could not be constructed out of empty proto object" },
-
-        EmptyVersions
-            | _ | { "empty supported versions" },
-
-        EmptyFeatures
-            | _ | { "empty supported features" },
-
-        NoCommonVersion
-            | _ | { "no common version" },
-
-        VersionNotSupported
-            {
-                version: Version,
-            }
-            | e | { format_args!("version \"{}\" not supported", e.version) },
-
-        InvalidAddress
-            | _ | { "invalid address" },
-
-        MissingProofHeight
-            | _ | { "missing proof height" },
-
-        MissingConsensusHeight
-            | _ | { "missing consensus height" },
-
-        InvalidProof
-            [ ProofError ]
-            | _ | { "invalid connection proof" },
-
-        VerifyConnectionState
-            [ client_error::Error ]
-            | _ | { "error verifying connnection state" },
-
-        Signer
-            [ SignerError ]
-            | _ | { "invalid signer" },
-
-        ConnectionNotFound
-            { connection_id: ConnectionId }
-            | e | {
-                format_args!("no connection was found for the previous connection id provided {0}",
-                    e.connection_id)
-            },
-
-        InvalidCounterparty
-            | _ | { "invalid signer" },
-
-        ConnectionIdMismatch
-            {
-                connection_id: ConnectionId,
-                counterparty_connection_id: ConnectionId,
-            }
-            | e | {
-                format_args!("counterparty chosen connection id {0} is different than the connection id {1}",
-                    e.connection_id, e.counterparty_connection_id)
-            },
-
-        MissingCounterparty
-            | _ | { "missing counterparty" },
-
-
-        MissingCounterpartyPrefix
-            | _ | { "missing counterparty prefix" },
-        MissingClientState
-            | _ | { "missing client state" },
-
-        NullClientProof
-            | _ | { "client proof must be present" },
-
-        FrozenClient
-            { client_id: ClientId }
-            | e | {
-                format_args!("the client id does not match any client state: {0}",
-                    e.client_id)
-            },
-
-        ConnectionVerificationFailure
-            | _ | { "the connection proof verification failed" },
-
-        ConsensusStateVerificationFailure
-            { height: Height }
-            [ client_error::Error ]
-            | e | {
-                format_args!("the consensus proof verification failed (height: {0})",
-                    e.height)
-            },
-
+use displaydoc::Display;
+
+#[derive(Debug, Display)]
+pub enum ConnectionError {
```

Comment:
Here are the error variants that we still have that are unused:
+ `FrozenClient`
+ `ImplementationSpecific`

Reply:
done